src/crimson/osd/pg.cc

```diff
@@ -930,12 +949,11 @@ PG::with_clone_obc(hobject_t oid, with_obc_func_t&& func)
     [oid, func=std::move(func), this](auto head) -> load_obc_iertr::future<> {
     auto coid = resolve_oid(head->get_ro_ss(), oid);
     if (!coid) {
```

Comment:
This change to permit returning an object_corrupted should be in its own commit since it seems to be independent of the rest of the commit.